### ISSUE TITLE
Investigate partial splits in `config_split` 2.x

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#4035]
# Motivation

Partial splits have been introduced to the 2.x branch of the `config_split` module. Partial splits will allow defining only what is different in a piece of config instead of requiring a complete copy of it. This idea could be revolutionary for our platform, allowing us to capture a lot less redundant configuration for our splits and simplifying the maintenance of things like permissions.

# Proposal

Investigate how the new functionality works, how we can use it, and document what changes would need to happen to the way our splits and our workflow for setting them up. This functionality is brand new, so I don't expect that we'll be able to adopt it right away, but ideally the work on this issue could help move changes to `config_split` forward as well.

**Comment by pyrello** [comment]
I'm definitely squatting on this. Kidding. Not kidding. Kidding...

**Comment by pyrello** [comment]
One thing you'll notice if you look at what happens to a [`config_split.patch.user.role.*.yml` file](https://github.com/uiowa/uiowa/pull/4091/commits/b2f9d8847917aff4ca62637fe1055fffa9311979) is that the current array diff procedure picks up on a lot of false negative because the keys don't match. The `config_split` maintainer created an issue to track this problem: https://www.drupal.org/project/config_split/issues/3232243

**Comment by pyrello** [comment]
I have created an MR for the above issue. It seems like it solves the sequencing issue, but it breaks existing tests, so I'm waiting from some feedback from the maintainer before I can move forward with it.

**Comment by pyrello** [comment]
Another piece of feedback I had for the maintainer is that the way the `added` and `removed` keys are being used in the `config_split.patch.*` files is confusing because `added` is used to describe things that will not be there after the split patch is merged in and `removed` is used to describe things that will be there, which I found confusing. This issue/MR is an attempt to resolve that: https://www.drupal.org/project/config_split/issues/3232667

**Comment by pyrello** [comment]
# Conclusions of Investigation

## Limitations
As of right now, there are some outstanding issues that would need to be resolved to allow us to move forward with Config Split 2.x:
- [Config sequences](https://www.drupal.org/docs/drupal-apis/configuration-api/configuration-schemametadata#s-sequence-definitions-) are not handled correctly in at least a couple critical cases: `user.role.*:permissions`, `dependencies` in any config that uses them. Work is ongoing to fix this: https://www.drupal.org/project/config_split/issues/3232243
- Reading the new `config_split.patch.*` format is confusing because it shows items being added and removed as the opposite as what is actually happening when you do an import. Work is ongoing to fix this: https://www.drupal.org/project/config_split/issues/3232667
- Current diff methods are not handling serialization of LB settings in `core.entity_view_display.node.*.default` correctly. There is an open issue to track this: https://www.drupal.org/project/config_split/issues/3231541

## Expected benefits
- Making this change will dramatically simplify the maintenance of several problematic config types including `user.role.*`, `core.entity_view_display.**`, and `metatag.settings`. Here are some examples of before and after:
  * Webmaster permissions for Areas of Study
    **Before:** https://github.com/uiowa/uiowa/blob/main/config/features/areas_of_study/user.role.webmaster.yml
    **After:** https://github.com/uiowa/uiowa/blob/901edeee502b4cab6219e9b55849335a8f1bb115/config/features/areas_of_study/config_split.patch.user.role.webmaster.yml
  * Metatag settings for Areas of Study
    **Before:** https://github.com/uiowa/uiowa/blob/main/config/features/areas_of_study/metatag.settings.yml
    **After:** https://github.com/uiowa/uiowa/blob/901edeee502b4cab6219e9b55849335a8f1bb115/config/features/areas_of_study/config_split.patch.metatag.settings.yml
- The above changes will also make it easier to combine multiple splits together with a minimum of redundancy.
- This all results in smaller config files for config that can be put in the patch format, which results in less config overall in the project. This means less technical debt!

## Recommendations
- Finish the work that can be done now without conflicts.
- Continue to work with the maintainer of Config Split to resolve above issues as well as anything else that may be a problem that we have not figured out yet.
- Figure out updated workflow and update any documentation.
- Automate basic process of creating splits for custom/new content types?

**Comment by pyrello** [comment]
Closing this in favor of #4237
